Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h

```diff
@@ -643,6 +651,8 @@ class WebsiteDataStore : public API::ObjectImpl<API::Object::Type::WebsiteDataSt
 #if ENABLE(CONTENT_EXTENSIONS)
     RefPtr<WebCore::ResourceMonitorThrottler> m_resourceMonitorThrottler;
 #endif
+
+    unsigned m_ongoingRemoveDataTasks { 0 };
```

Comment:
This kind of counter is too easy to get out of sync. If we get it wrong, we could end up disabling the process cache entirely. My recommendation would be to use WTF::RefCounter, and capture a RefCounter::Token in each remove task lambda